packages/components/src/spinner/index.js

```diff
@@ -1,8 +1,54 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+
 /**
  * Internal dependencies
  */
-import { StyledSpinner } from './styles/spinner-styles';
+import { COLORS, CONFIG } from '../utils';
+
+/**
+ * WordPress dependencies
+ */
+import { SVG, Circle } from '@wordpress/primitives';
+
+// A dedicated wrapper allows us to apply width and shadow color based on config values
+// The SVG can only have a centered stroke so a stacked box-shadow accomplishes a matching 1.5px border
+export const StyledSpinner = styled.span`
+	width: ${ CONFIG.spinnerSize }px;
+	height: ${ CONFIG.spinnerSize }px;
+	box-shadow: inset 0 0 0 0.75px ${ COLORS.gray[ 300 ] },
+		0 0 0 0.75px ${ COLORS.gray[ 300 ] };
+`;
 
 export default function Spinner() {
-	return <StyledSpinner className="components-spinner" />;
+	const viewBox = `0 0 ${ CONFIG.spinnerSize } ${ CONFIG.spinnerSize }`;
+	const radius = Number( CONFIG.spinnerSize ) / 2;
+	const strokeDasharray = [
+		Number( CONFIG.spinnerSize ),
+		Number( CONFIG.spinnerSize ) * 10,
+	];
```

Comment:
Sorry if missed this while skimming through the previous conversations, but is there a reason why we compute dynamically `viewBox`, `radius` and `strokeDasharray`, all in relation to `spinnerSize` ?

With SVG we should be able to define an arbitrary `viewBox` and then scale everything accordingly. We could then use `spinnerSize` in the styles to change `width` and `height` of the `SVG` element itself ([example](https://codepen.io/ciampo/pen/RwLmgPo))

Reply:
Yeah that would be a lot simpler wouldn't it :) We'll need to keep the computed `stroke-dasharray` though in order for the indicator to scale correctly.